### PR TITLE
fix(workspace-store): keep dots in webhook deep-link slugs

### PR DESCRIPTION
**Pull request by hanspagel**
Webhook event names with dots (like `account_holder.created`) had the dot deleted when building the navigation link, joining the words into `account-holdercreated`. Now the dot is kept, so the link reads `account-holder.created`.

Before / after:

| Webhook name | Before | After |
|---|---|---|
| `account_holder.created` | `account-holdercreated` | `account-holder.created` |
| `card_transaction.updated` | `card-transactionupdated` | `card-transaction.updated` |
| `tokenization.two_factor_authentication_code` | `tokenizationtwo-factor-authentication-code` | `tokenization.two-factor-authentication-code` |

Underscores still become hyphens; only the dot is preserved. The slug change only touches the webhook branch, so tag and model links are unchanged.

## Redirects

Renaming the slug would break old bookmarks (`.../webhook/POST/account-holdercreated`), so old links now redirect to the new slug. The old slug dropped the dot and cannot be rebuilt from a pattern, so the redirect is built from the real webhook names in the document (legacy slug -> current slug per webhook) in `buildWebhookRedirects` inside `packages/api-reference/src/helpers/id-routing.ts`, reusing the existing id-space redirect layer.

Verified in the browser: loading a legacy `.../webhook/POST/account-holdercreated` link rewrites the URL to `.../webhook/POST/account-holder.created` and scrolls to the webhook.

## Tests

- `get-navigation-options`: webhook slugs keep dots, underscores still become hyphens.
- `slugify`: pins what a dot does by default.
- `id-routing`: legacy webhook slug redirects to the current one (hash, path routing, tag prefix, sub-anchor, and no-op cases).

---

> [!NOTE]
> **Low Risk**
> Localized to webhook URL slug generation and client-side URL canonicalization; collision/ambiguity guards limit wrong redirects.
> 
> **Overview**
> **Webhook deep links** now keep dots when slugifying event names (e.g. `account_holder.created` → `account-holder.created` instead of `account-holdercreated`). Only the default webhook ID path in `get-navigation-options` changes; custom `generateWebhookSlug` and tag/model IDs are unchanged.
> 
> **Backward compatibility:** `redirectUrl` gains optional webhook metadata and `buildWebhookRedirects`, which maps each webhook’s old dot-stripped slug to the new one for hash, path, and tagged URLs, including property anchors (`.body.id`, `.responses.…`). Redirects are skipped when the legacy slug is ambiguous or already a valid current slug for another webhook. **ApiReference** collects webhooks from loaded navigation and runs this redirect after the document loads so the URL and scroll target update before scrolling.
> 
> Tests cover navigation ID generation, default `slugify` dot behavior, and redirect edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 19bfec602237773034e3410f4f4175a57073b1b5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/workspace-store/src/navigation/get-navigation-options.ts [hunk 1]
```diff
@@ -121,7 +121,10 @@ export const getNavigationOptions = (documentName: string, options?: NavigationO
         })}`
       }
 
-      return `${prefixTag}webhook/${props.method?.toUpperCase()}/${slugify(props.name)}`
+      // Webhook events are commonly named with dots (e.g. "account_holder.created").
+      // Keep the dot so the deep link stays close to the event name, instead of
+      // dropping it and joining adjacent words ("account-holdercreated").
+      return `${prefixTag}webhook/${props.method?.toUpperCase()}/${slugify(props.name, { allowedSpecialChars: '.' })}`
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/workspace-store/src/navigation/get-navigation-options.ts:127]: Addressed in 359440f: legacy webhook deep links now redirect to the current slug via `buildWebhookRedirects` in `id-routing.ts`, built from the real webhook names since the dropped-dot slug cannot be reversed generically. Covered by unit tests and verified in the browser.

File: packages/api-reference/src/helpers/id-routing.ts [hunk 2]
```diff
@@ -321,13 +323,87 @@ const buildRedirects = ({ modelsSectionSlug, documentSlug, isMultiDocument }: Re
...
+      currentKeys.has(legacyKey) ||
+      (legacyCounts.get(legacyKey) ?? 0) > 1
+    ) {
+      continue
+    }
+
+    redirects.push({
+      match: new RegExp(`(^|/)webhook/${escapeRegex(upperMethod)}/${escapeRegex(legacySlug)}(?=$|/)`),
+      replace: (_match, prefix) => `${prefix}webhook/${upperMethod}/${currentSlug}`,
+    })
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/api-reference/src/helpers/id-routing.ts:410]: Addressed in 19bfec6: the slug boundary now also accepts the dot-joined schema-param markers (`.body.`, `.path.`, `.query.`, `.header.`, `.responses.`), so legacy links like `…/account-holdercreated.body.id` redirect to `…/account-holder.created.body.id`. A dot followed by anything else is still not a boundary, since a current webhook slug can legitimately contain dots. Covered by new unit tests (body anchor, responses anchor, and a non-marker dot suffix staying untouched).

**Review by marclave — APPROVED** [review]
